### ISSUE TITLE
mismatch for docs in `perplexity`

### ISSUE THREAD

**Issue by leonardodalinky** [Lightning-AI/torchmetrics#1698]
## 📚 Documentation

I am working on the NLP models and find that the implementation of `perplexity` in https://github.com/Lightning-AI/torchmetrics/blob/e4df07109586d1f605e06aed85fbc6794e8ed833/src/torchmetrics/functional/text/perplexity.py#L114-L126 is probably wrong.

Since `_perplexity_update` in https://github.com/Lightning-AI/torchmetrics/blob/e4df07109586d1f605e06aed85fbc6794e8ed833/src/torchmetrics/functional/text/perplexity.py#L84-L88 execute the `softmax`, it indicates that the `preds` **should be a logit or a unnormalized score** and definitely NOT a log probability.

I wonder if this is intentioned move or is just a simple typo. 

I also find the similar fix in issue #1366, which made the mistake. Should I make a PR to this?

**Comment by gortibaldik** [comment]
Hi, are you sure this issue is resolved ?

currently the documentation states the following:
```
"""Compute intermediate statistics for Perplexity.

    Args:
        preds:
            Probabilities assigned to each token in a sequence with shape [batch_size, seq_len, vocab_size].
        target:
            Ground truth values with a shape [batch_size, seq_len].
        ignore_index:
            Integer specifying a target class to ignore. If given, this class index does not contribute
            to the returned score.

    Returns:
        Log probabilities, summed over all samples
        Number of samples
    """
```
    
```
probs = F.softmax(preds.reshape(-1, preds.shape[-1]), dim=1)
```
And again, softmax applied to probabilities and softmax applied to unnormalized scores are two really different things !
